Client/Catalog/frontend/store/configuration/configuration.effects.ts

```diff
@@ -438,15 +438,29 @@ export class ConfigurationEffects {
 					};
 				}
 
-				return this.configurationRepository.addToBasket({
-					productId: configurationState.productId,
-					locale: configurationState.connector?.locale,
-					compressedState: configurationState.state.compressedState,
-					quantity: configurationState.quantity,
-					perspectives: configurationState.perspectives,
-					sessionCookies: configurationState.connector?.sessionCookies,
-					additionalData
-				});
+        // this is the case when user from the basket clicks "Konfiguration bearbeiten" and then changes the config and adds again to the basket
+        if (payload.configurationId) {
```

Comment:
I think we have a second condition here. Not only if the configurationId is set, we must also check if we are in basket route. Because if we are in offer route we can not update a basket item. So we need to check  configurationType==='basket' route parameter too. Maybe we can symply add configurationType as optional parameter to the action too.

Reply:
done changes